src/replication.c

```diff
@@ -1568,14 +1568,17 @@ void rdbPipeReadHandler(struct aeEventLoop *eventLoop, int fd, void *clientData,
 
         if (stillAlive == 0) {
             serverLog(LL_WARNING,"Diskless rdb transfer, last replica dropped, killing fork child.");
+            aeDeleteFileEvent(server.el, server.rdb_pipe_read, AE_READABLE);
             killRDBChild();
```

Comment:
don't we have to add a `break;` to this section now?

Reply:
ohh, you help me find the problem https://github.com/redis/redis/pull/13410/files#r1677593292.

Reply:
not sure what you mean.
am i right that a break should be added here?

Reply:
yes, we need a `break` here.